src/internal/m365/collection/groups/collection.go

```diff
@@ -370,6 +378,44 @@ func (col *lazyFetchCollection[C, I]) streamItems(ctx context.Context, errs *fau
 				"item_id", id,
 				"parent_path", path.LoggableDir(col.LocationPath().String()))
 
+			// Handle metadata before data so that if metadata file fails,
+			// we are not left with an orphaned data file.
+			//
+			// If the data download fails for some reason other than deleted in
+			// flight, we will still end up persisting a .meta file. This is
+			// fine however, since the next backup will overwrite it.
+			//
+			// If item is deleted in flight, we will end up with an orphaned
+			// .meta file. The only impact here is storage bloat, which
+			// is minimal. Other impact could be if we do an in-order restore
+			// using a tree built from .meta files. We may have some .meta
+			// files without corresponding .data files.
```

Comment:
if restore is requesting meta files based on translated paths that were sourced from backup details then I think we'll actually be ok. The reason for that is that if we failed to get the data file the item won't appear in backup details, thus we won't translate the path for the meta file and will essentially act as if it doesn't exist